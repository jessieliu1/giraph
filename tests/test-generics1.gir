int main() {
	graph<int> g = [A:3 -- B:4];
	for_node(n : g) {
		print(n.data());
	}

	graph<string> g2 = [C:"hey" -- D:"you"];
	for_node(n : g2) {
		print(n.data());
	}

	graph<float> g3 = [E:1.0 -- F:2.0];
	for_node(n : g3) {
		if (n.data() == 1.0) {
			prints("E");
		} else if (n.data() == 2.0) {
			prints("F");
		}
	}

	graph<graph<string>> g4 = [G:g2];
	for_node(ng : g4) {
		for_node(n : ng.data()) {
			prints(n.data());
		}
	}
	return 0;
}