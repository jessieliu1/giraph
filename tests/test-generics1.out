3
4
hey
you
E
F
hey
you
